source/js/buyers-guide/search/sort-creepiness.js

```diff
@@ -0,0 +1,151 @@
+const SPACEBAR_KEY_CODE = [0, 32];
+const ENTER_KEY_CODE = 13;
+const DOWN_ARROW_KEY_CODE = 40;
+const UP_ARROW_KEY_CODE = 38;
+const ESCAPE_KEY_CODE = 27;
+
+export class PNISortCreepiness {
```

Comment:
Can you add some context as to where this class is doing the sorting? I seem to miss it.

Reply:
L90 `this.searchFilter.updateSort(e.target.dataset.value);`

Reply:
So `PNISortCreepiness` uses `SearchFilter.updateSort` which uses `CreepUtils.sortOnCreepiness`? Looks like this is also the only use of `SearchFilter` in `PNISortCreepiness`. Why do we need to go through `SearchFilter.updateSort`?

Reply:
I renamed PNISortCreepiness to PNISortDropdown since I think the naming made it harder to read before. 

`CreepUtils.sortOnCreepiness` does not update the history so we need a function that calls the sort method and update history.state. We could do it in PNISortDropdown but since it was similar to some methods of updating the category, I moved it to the searchFilter class + CreepUtils.sortOnCreepiness is not imported outside of the search-filter instance.

Reply:
Since it looks like `SearchFilter.updateSort` does not reference `this` or any other method or attribute of the `SearchFilter`, what do you thing about moving the rest of the logic to the `sortOnCreepiness` method?

Reply:
I am also wondering, why are we using classes for these methods? Many seem to be `static` and not reference `this` or are even used on an instance. I am probably missing something, but why are we not using functions?

Reply:
`sortOnCreepiness` is used multiple times on other files where they are already using `pushState` or `replaceState`. If we move the rest of the logic to `sortOnCreepiness` then we might be doing 2 updates to the browser history in a row and might make debugging future history issues more difficult.

Reply:
If you want more details about why we choose classes vs functions for PNI JS files, Pomax did majority of the refactor last year: https://github.com/mozilla/foundation.mozilla.org/pull/8007 He also made a todo list that gave the steps on, `source/js/buyers-guide/search/continue.md`. It could be preference but for me I am just following a similar pattern of what is already written for the search.js related files.

Reply:
Sorry, I don't see where #8007 explains the reason for classes over function. I see that this introduced it 👍

Reply:
> I am open to changing the name of updateSort to something that reflects the task better, but I do want a function to exist for this use case.

I like the idea of renaming it to better reflect what it does. 

However, I am also wondering if it would make sense to also move `updateSort` to the `PNISortDropdown` since this is what does the sorting. 

What are the other functions that need to update the sort? I was my impression that changing the sort order was a new feature we are adding.

Reply:
`performInitialHistoryReplace() clearText() filterCategory() applyHistory()` all use the `sortOnCreepiness()` so we should not manipulate `sortOnCreepiness()`. 

`updateSort()` does rely on `searchFilter.categoryTitle gsap and Utils.getTitle()`, we could do that in the PNISortDropdown but that would involve a few additional imports which I think makes less sense at the moment.

I think the names of the functions do not really represent the new changes so I will change `updateSort -> updateSortHistoryState` and `sortOnCreepiness -> sortProductCards`.

Reply:
Thanks for renaming functions @fessehaye 

I still think there is some opportunity to refactor this. The interdependencies between all the modules are still quite nebulous to me. But, in light of the deadline coming closer we have to do it another time...